### PR TITLE
CR-1145195 & CR-1145192 xgq: retrieve clock scaling configs properly

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Signed-off-by: Rajkumar Rampelli <rampelli@amd.com>

#### Problem solved by the commit
xgq driver requesting for clock scaling configs only once during it's init sequence and copied to scaling payload. But, this payload getting override by other sensor payload request/response data since all payloads are combined with union. So, at a time, only one payload information is valid. Fix is to request clock scaling configs whenever required in xgq driver and read the configs from the scaling payload.
Change#2: When XRT drivers are reloaded, during init, XRT reads previously configured clock scaling values, but not the default values. To fix this problem, introduced a new clock scaling command option "reset" for resetting clock scaling configuration parameters to default settings. So, XRT sets reset option and issues scaling command for reading default settings.
Change#3: By default, power scaling override & temperature scaling override are disabled. if user tries to set override values, then only enable override flag. So, in xbmgmt cmc report, override enable flag shows disabled now by default.
VMR PR https://github.com/Xilinx/VMR/pull/265 
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
CR-1145195 & CR-1145192
#### How problem was solved, alternative solutions (if any) and why they were rejected
Added changes in XGQ driver to issue clock scaling configs read request to VMR wherever required.
#### Risks (if any) associated the changes in the commit
NA
#### What has been tested and how, request additional testing if necessary
xbmgmt's cmc report & config commands, and access xgq's scaling sysfs nodes
#### Documentation impact (if any)
NA

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
VMR PR https://github.com/Xilinx/VMR/pull/265 is also under review.

**Comment by rajkumar-xilinx** [comment]
Hi all, 
VMR changes are merged https://github.com/Xilinx/VMR/pull/265 . 
Please review this PR, thank you.

**Review by chvamshi-xilinx — APPROVED** [review]

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c [hunk 1]
```diff
@@ -2487,18 +2521,25 @@ static ssize_t clk_scaling_stat_raw_show(struct device *dev,
 	struct xgq_cmd_cq_clk_scaling_payload *cs_payload=
 		(struct xgq_cmd_cq_clk_scaling_payload *)&xgq->xgq_cq_payload;
 	ssize_t cnt = 0;
+	int ret = 0;
+
+	ret = clk_scaling_status_query(xgq->xgq_pdev);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c:2526]: don't you need to grab the lock (lock &xgq->clk_scaling_lock)?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c [hunk 2]
```diff
@@ -2547,6 +2596,12 @@ static ssize_t clk_scaling_configure_store(struct device *dev,
 	char* end = (char*) buf;
 	int ret = 0;
 
+	ret = clk_scaling_status_query(xgq->xgq_pdev);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c:2599]: don't you need to grab the lock (lock &xgq->clk_scaling_lock)?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c [hunk 3]
```diff
@@ -2609,9 +2661,18 @@ static ssize_t xgq_scaling_temp_override_show(struct device *dev,
                                               char *buf)
 {
 	struct xocl_xgq_vmr *xgq = platform_get_drvdata(to_platform_device(dev));
+	struct xgq_cmd_cq_clk_scaling_payload *cs_payload =
+		(struct xgq_cmd_cq_clk_scaling_payload *)&xgq->xgq_cq_payload;
 	ssize_t cnt = 0;
+	int ret = 0;
 
-	cnt += sprintf(buf + cnt, "%u\n", xgq->temp_scaling_ovrd_limit);
+	ret = clk_scaling_status_query(xgq->xgq_pdev);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c:2669]: don't you need to grab the lock (lock &xgq->clk_scaling_lock)?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c [hunk 4]
```diff
@@ -2626,23 +2687,29 @@ static ssize_t xgq_scaling_temp_override_store(struct device *dev,
 	u16 temp = 0;
 	int ret = 0;
 
+	ret = clk_scaling_status_query(xgq->xgq_pdev);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c:2690]: don't you need to grab the lock (lock &xgq->clk_scaling_lock)?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c [hunk 5]
```diff
@@ -2655,9 +2722,18 @@ static ssize_t xgq_scaling_power_override_show(struct device *dev,
                                                char *buf)
 {
 	struct xocl_xgq_vmr *xgq = platform_get_drvdata(to_platform_device(dev));
+	struct xgq_cmd_cq_clk_scaling_payload *cs_payload =
+		(struct xgq_cmd_cq_clk_scaling_payload *)&xgq->xgq_cq_payload;
 	ssize_t cnt = 0;
+	int ret = 0;
 
-	cnt += sprintf(buf + cnt, "%u\n", xgq->pwr_scaling_ovrd_limit);
+	ret = clk_scaling_status_query(xgq->xgq_pdev);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c:2730]: don't you need to grab the lock (lock &xgq->clk_scaling_lock)?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c [hunk 6]
```diff
@@ -2672,23 +2748,29 @@ static ssize_t xgq_scaling_power_override_store(struct device *dev,
 	u16 pwr = 0;
 	int ret = 0;
 
+	ret = clk_scaling_status_query(xgq->xgq_pdev);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c:2751]: don't you need to grab the lock (lock &xgq->clk_scaling_lock)?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c [hunk 7]
```diff
@@ -2704,6 +2786,13 @@ static ssize_t xgq_scaling_enable_show(struct device *dev,
 	struct xgq_cmd_cq_clk_scaling_payload *cs_payload =
 		(struct xgq_cmd_cq_clk_scaling_payload *)&xgq->xgq_cq_payload;
 	ssize_t cnt = 0;
+	int ret = 0;
+
+	ret = clk_scaling_status_query(xgq->xgq_pdev);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c:2791]: don't you need to grab the lock (lock &xgq->clk_scaling_lock)?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c [hunk 8]
```diff
@@ -2721,6 +2810,12 @@ static ssize_t xgq_scaling_enable_store(struct device *dev,
 	u32 val = 0;
 	int ret = 0;
 
+	ret = clk_scaling_status_query(xgq->xgq_pdev);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c:2813]: don't you need to grab the lock (lock &xgq->clk_scaling_lock)?

**Review by maxzhen — CHANGES_REQUESTED** [review]

**Comment by rajkumar-xilinx** [comment]
Hi @maxzhen 
I have taken care of your review comments, could you please review it again and let me know? Thank you.

**Review by maxzhen — APPROVED** [review]

**Comment by rajkumar-xilinx** [comment]
retest this please

**Pull request merged**